### ISSUE TITLE
Catching null-based expectations in test code.

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/linter#4801]
I just saw this in some test code: `expect(context.analysisOptions, isNotNull);`. This seemed fine until I realized that `context.analysisOptions` returns a non-nullable value making it impossible for the condition to ever fail. I wonder whether it would be useful to have a lint to catch places where `isNull`, `isNotNull`, or `null` are used in expectations where they can never fail.

**Comment by srawlins** [comment]
Yeah that would make a great lint the test package could provide in an analyzer plugin 😁
